microcosm/tests/config/check_warnings.py

```diff
@@ -35,10 +35,11 @@ def check_unsupported_arg_warning():
 
 @contextmanager
 def check_no_warnings():
-    with warnings.catch_warnings(record=True) as w:
+    with warnings.catch_warnings(record=True) as caught_warnings:
         warnings.simplefilter("always")
 
         yield
 
-        if w:
-            assert_that(w, is_(empty()), w[-1].message)
+        caught_warnings = [warning for warning in caught_warnings if "virtualenvs" not in warning.filename]
```

Comment:
Crude way of excluding warning for externals libraries.

Reply:
What was being thrown here for you?

Reply:
```{message : DeprecationWarning("Using or importing the ABCs from \'collections\' instead of from \'collections.abc\' is deprecated, and in 3.8 it will stop working"), category : \'DeprecationWarning\', filename : \'/Users/adrienfallou/.virtualenvs/microcosm/lib/python3.7/site-packages/tornado/httputil.py\', lineno : 107, line : None}```